processing/src/main/java/io/druid/query/search/search/SearchSortSpec.java

```diff
@@ -19,22 +19,83 @@
 
 package io.druid.query.search.search;
 
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+import io.druid.query.ordering.StringComparator;
+import io.druid.query.ordering.StringComparators;
 
 import java.util.Comparator;
 
-/**
- */
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = LexicographicSearchSortSpec.class)
-@JsonSubTypes(value = {
-    @JsonSubTypes.Type(name = "lexicographic", value = LexicographicSearchSortSpec.class),
-    @JsonSubTypes.Type(name = "alphanumeric", value = AlphanumericSearchSortSpec.class),
-    @JsonSubTypes.Type(name = "strlen", value = StrlenSearchSortSpec.class)
-})
-public interface SearchSortSpec
+public class SearchSortSpec
 {
-  Comparator<SearchHit> getComparator();
+  public static final StringComparator DEFAULT_ORDERING = StringComparators.LEXICOGRAPHIC;
+
+  private final StringComparator ordering;
+
+  @JsonCreator
+  public SearchSortSpec(
+      @JsonProperty("ordering") StringComparator ordering
```

Comment:
I think this needs to be called `"type"` not `"ordering"` in order for this to be backwards compatible.